server/src/metakb/query.py

```diff
@@ -199,196 +196,129 @@ async def search_statements(
         :return: A normalized therapy concept if it exists
         """
-        _, normalized_therapy_id = self.vicc_normalizers.normalize_therapy(therapy)
-
-        if not normalized_therapy_id:
-            warnings.append(f"Therapy Normalizer unable to normalize: {therapy}")
-        return normalized_therapy_id
-
-    def _get_normalized_disease(self, disease: str, warnings: list[str]) -> str | None:
-        """Get normalized disease concept.
-
-        :param disease: Disease query
-        :param warnings: A list of warnings for the search query
-        :return: A normalized disease concept if it exists
-        """
-        _, normalized_disease_id = self.vicc_normalizers.normalize_disease(disease)
-
-        if not normalized_disease_id:
-            warnings.append(f"Disease Normalizer unable to normalize: {disease}")
-        return normalized_disease_id
+        _, therapy_id = self.vicc_normalizers.normalize_therapy(therapy)
+        return SearchTerm(
+            term=therapy, term_type=SearchTermType.THERAPY, resolved_id=therapy_id
+        )
 
-    async def _get_normalized_variation(
-        self, variation: str, warnings: list[str]
-    ) -> str | None:
+    async def _get_normalized_variation(self, variation: str) -> SearchTerm:
         """Get normalized variation concept.
 
         :param variation: Variation query
-        :param warnings: A list of warnings for the search query
         :return: A normalized variant concept if it exists
         """
         variant_norm_resp = await self.vicc_normalizers.normalize_variation(variation)
         normalized_variation = variant_norm_resp.id if variant_norm_resp else None
 
-        if not normalized_variation:
-            # Check if VRS variation (allele, copy number change, copy number count)
-            if variation.startswith(("ga4gh:VA.", "ga4gh:CX.", "ga4gh:CN.")):
-                normalized_variation = variation
-            else:
-                warnings.append(
-                    f"Variation Normalizer unable to normalize: {variation}"
-                )
-        return normalized_variation
-
-    def _get_normalized_gene(self, gene: str, warnings: list[str]) -> str | None:
-        """Get normalized gene concept.
-
-        :param gene: Gene query
-        :param warnings: A list of warnings for the search query.
-        :return: A normalized gene concept if it exists
-        """
-        _, normalized_gene_id = self.vicc_normalizers.normalize_gene(gene)
-        if not normalized_gene_id:
-            warnings.append(f"Gene Normalizer unable to normalize: {gene}")
-        return normalized_gene_id
+        # Check if VRS variation (allele, copy number change, copy number count)
+        # TODO should this happen before the normalizer call?
+        if not normalized_variation and variation.startswith(
+            ("ga4gh:VA.", "ga4gh:CX.", "ga4gh:CN.")
+        ):
+            normalized_variation = variation
```

Comment:
Yes, I do think if they're passing a VRS ID (do we want or need to support this?), we should do this first and not attempt to normalize.